token/services/certifier/interactive/client.go

```diff
@@ -98,95 +107,141 @@ func (d *CertificationClient) RequestCertification(ids ...*token.ID) error {
 		return nil
 	}
 
-	resultBoxed, err := d.viewManager.InitiateView(NewCertificationRequestView(d.channel, d.namespace, d.certifiers[0], toBeCertified...))
+	var resultBoxed interface{}
+	var err error
+	for i := 0; i < 3; i++ {
+		resultBoxed, err = cc.viewManager.InitiateView(NewCertificationRequestView(cc.channel, cc.namespace, cc.certifiers[0], toBeCertified...))
+		if err != nil {
+			logger.Errorf("failed to request certification, try again [%d]...", i)
+			continue
+		}
+		break
+	}
 	if err != nil {
 		return err
 	}
 	certifications, ok := resultBoxed.(map[*token.ID][]byte)
 	if !ok {
 		return errors.Errorf("invalid type, expected map[token.ID][]byte")
 	}
-	if err := d.certificationStorage.Store(certifications); err != nil {
+	if err := cc.certificationStorage.Store(certifications); err != nil {
 		return err
 	}
 	return nil
 }
 
-func (d *CertificationClient) Start() error {
-	go d.Scan()
-	return nil
-}
+func (cc *CertificationClient) Scan() error {
+	logger.Debugf("check the certification of unspent tokens...")
+	// Check the unspent tokens
+	var err error
+	tokens, err := cc.queryEngine.UnspentTokensIterator()
+	if err != nil {
+		return errors.WithMessagef(err, "failed to get an iterator over unspent tokens")
+	}
+	defer tokens.Close()
 
-func (d *CertificationClient) Scan() {
-	var lastTXID string
-	var tokens driver.UnspentTokensIterator
+	var toBeCertified []*token.ID
 	for {
-		if tokens != nil {
-			tokens.Close()
-		}
-
-		logger.Debugf("check the certification of unspent tokens...")
-		// Check the unspent tokens
-		var err error
-		tokens, err = d.queryEngine.UnspentTokensIterator()
+		token, err := tokens.Next()
 		if err != nil {
-			logger.Errorf("failed to get an iterator over unspent tokens, wait and retry [%s]", err)
-			time.Sleep(d.waitTime)
-			continue
+			logger.Errorf("failed to get next unspent tokens, stop here [%s]", err)
+			break
 		}
-
-		var toBeCertified []*token.ID
-		for {
-			token, err := tokens.Next()
-			if err != nil {
-				logger.Errorf("failed to get next unspent tokens, stop here [%s]", err)
-				break
-			}
-			if token == nil {
-				break
-			}
-
-			// does token have a certification?
-			if !d.certificationStorage.Exists(token.Id) {
-				// if no, batch it
-				toBeCertified = append(toBeCertified, token.Id)
-			}
+		if token == nil {
+			break
 		}
-		tokens.Close()
-
-		if len(toBeCertified) != 0 {
-			// Request certification
-			logger.Debugf("request certification of [%v]", toBeCertified)
-			if err := d.RequestCertification(toBeCertified...); err != nil {
-				logger.Errorf("failed retrieving certification [%s], try later", err)
-				time.Sleep(2 * time.Second)
-				continue
-			}
-			logger.Debugf("request certification of [%v] satisfied with no error", toBeCertified)
+
+		// does token have a certification?
+		if !cc.certificationStorage.Exists(token.Id) {
```

Comment:
We could Duse `cc.IsCertified`

Reply:
is it just to avoid a call to another function 😓